internal/crd_cache.go

```diff
@@ -0,0 +1,29 @@
+package internal
+
+import (
+	"sync"
+
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type CustomResourceDefinitionCache struct {
+	storage *sync.Map
+}
+
+func NewCustomResourceDefinitionCache() *CustomResourceDefinitionCache {
+	return &CustomResourceDefinitionCache{storage: &sync.Map{}}
+}
+
+func (cache *CustomResourceDefinitionCache) Get(key client.ObjectKey) *v1.CustomResourceDefinition {
```

Comment:
`client.ObjectKey` is not suitable here, you don't need Name and Namespace to distinguish with the CRD (CRD is also not namespaced). The Key here can be CRD name only, a pure string is enough.